### PR TITLE
Modify csconfig.py checks to allow for N certs with same nickname

### PR DESCRIPTION
**Pull request by ckelleyRH**
Currently the base64 blob from nssdb is directly compared with a cert from the CA subsystem, this fails if there are multiple certs with the same nickname. As this is an allowable state, the tests are modified to break the base64 blob into individual certs.

Resolves: #2022561

### CODE REVIEW

File: base/server/healthcheck/pki/server/healthcheck/meta/csconfig.py [hunk 1]
```diff
@@ -55,22 +56,26 @@ def compare_nssdb_with_cs(class_instance, subsystem, cert_tag):
...
-                      msg='Certificate \'%s\' does not match the value '
-                          'of %s in %s' % (cert['nickname'],
-                                           directive,
-                                           subsystem.cs_conf))
-    else:
-        return Result(class_instance, constants.SUCCESS,
-                      key=cert_id,
-                      configfile=subsystem.cs_conf)
+    # base64 blob may contain multiple certs so, compare each in turn
+    cert_split = re.split(b'(Certificate:\n)', cert_nssdb)
```
Comment (ckelleyRH) [inline base/server/healthcheck/pki/server/healthcheck/meta/csconfig.py:60]: I'll explain what it happening here. If there are 2 certs with the same nickname, this regex will split `cert_nssdb` into 5 parts:

0. Empty string, so we can ignore this
1. "Certificate:\n"
2. <The rest of the first cert>
3. "Certificate:\n"
4. <The rest of the second cert>

So it is necessary to recombine the capture group with the rest of the cert. It is a bit dirty but I couldn't think of a more robust way of doing it - if you can I would be pleased to hear it!

**Comment by ckelleyRH** [comment]
@rcritten - as I said in the bug I'm not sure the best way to test, but I have the patch built into my COPR so I can plug it into the IPA CI if we want to make a test for this: https://copr.fedorainfracloud.org/coprs/ckelley/pki/builds/

**Comment by ckelleyRH** [comment]
I didn't notice that the default `certutil` output was PEM from this method rather than pretty print, I will fix in my fork before pushing here again as I expect there may be a second issue too.

File: base/server/healthcheck/pki/server/healthcheck/meta/csconfig.py (hunk 1 above)
Comment (ckelleyRH) [reply base/server/healthcheck/pki/server/healthcheck/meta/csconfig.py]: I took a different approach, it was easier to do in PEM format. It should be clearer as a result.

**Review by fmarco76 — APPROVED** [review]
LGTM

**Comment by ckelleyRH** [comment]
Thanks @fmarco76 !

**Pull request merged**